app/src/main/java/com/cornellappdev/android/eatery/MenuActivity.java

```diff
@@ -237,17 +265,30 @@ public void setPrimaryItem(ViewGroup container, int position, Object object) {
     public Fragment getItem(int position) {
       Bundle b = new Bundle();
       b.putInt("position", position);
-      b.putSerializable("cafeData", cafeData.getWeeklyMenu().get(cafeData.indexOfCurrentDay()));
+
+      if (mEatery instanceof DiningHallModel) {
+        b.putSerializable("mEatery", new ArrayList<>(((DiningHallModel) mEatery).getWeeklyMenu()
+            .get(ZonedDateTime.now().getDayOfWeek())));
+      }
+
       MenuFragment f = new MenuFragment();
       f.setArguments(b);
       return f;
     }
 
     @Override
     public int getCount() {
+
       int n;
+
       try {
-        n = cafeData.getWeeklyMenu().get(cafeData.indexOfCurrentDay()).size();
+        if (mEatery instanceof DiningHallModel) {
+          n = ((DiningHallModel) mEatery).getWeeklyMenu()
+              .get(ZonedDateTime.now().getDayOfWeek())
+              .size();
+        } else {
+          n = 0;
+        }
```

Comment:
if `n=0` in the `else` and in the `catch`, it can just be set to `0` before the `try`

Reply:
Java doesn't initialize local variables, so this was initializing it in all cases. I'll move the initialization up to the variable for clarity's sake.